docs/api-guide.md

```diff
@@ -1197,6 +1197,90 @@ In the sample code, the transaction is retried three times maximum and sleeps fo
+
+To enable the group commit feature, add the following configuration:
+
+#### Configurations
+
+```properties
+# By default, this configuration is set to `false`.
+scalar.db.consensus_commit.coordinator.group_commit.enabled=true
+
+# These properties are for the performance tuning of the group commit.
+# scalar.db.consensus_commit.coordinator.group_commit.group_size_fix_timeout_millis=40
+# scalar.db.consensus_commit.coordinator.group_commit.delayed_slot_move_timeout_millis=800
+# scalar.db.consensus_commit.coordinator.group_commit.old_group_abort_timeout_millis=30000
+# scalar.db.consensus_commit.coordinator.group_commit.timeout_check_interval_millis=10
+# scalar.db.consensus_commit.coordinator.group_commit.metrics_monitor_log_enabled=true
+```
+
+#### Limitations
+
+##### Custom transaction ID passed by users
+
+The group commit feature implicitly generates an internal value and uses it as a part of transaction ID. Therefore, a custom transaction ID manually passed by users via `com.scalar.db.transaction.consensuscommit.ConsensusCommitManager.begin(String txId)` or `com.scalar.db.transaction.consensuscommit.TwoPhaseConsensusCommitManager.begin(String txId)` can't be used as-is for later API calls. You need to use a transaction ID returned from`com.scalar.db.transaction.consensuscommit.ConsensusCommit.getId()` or `com.scalar.db.transaction.consensuscommit.TwoPhaseConsensusCommit.getId()` instead.
+
+```java
+   // This custom transaction ID needs to be used for ScalarDB transactions for some reason...
+   String myTxId = UUID.randomUUID().toString();
+
+   ...
+
+   DistributedTransaction transaction = manager.begin(myTxId);
+
+   ...
+
+   // When the group commit feature is enabled, a custom transaction ID passed by users can't be used as is.
+   // logger.info("The transaction state: {}", manager.getState(myTxId));
+   logger.info("The transaction state: {}", manager.getState(transaction.getId()));
+```
+
+##### Commit in Two-Phase Commit Interface
+
+The group commit feature manages all in-flight transactions in memory. When the feature is enabled with the two-phase commit interface, the information must be solely maintained by the coordinator service to prevent conflicts caused by participant services' inconsistent writes to the coordinator table, which may contain different transaction distributions over groups. Consequently, `com.scalar.db.transaction.consensuscommit.TwoPhaseConsensusCommit.commit()` must be called first by the coordinator service before being called by participant services.
```

Comment:
Is there a more accurate word we can use instead of `in-flight`, like `pending` or `in-progress` be accurate?

```suggestion
The group commit feature manages all in-flight transactions in memory. When the feature is enabled with the two-phase commit interface, the information must be solely maintained by the coordinator service to prevent conflicts caused by participant services' inconsistent writes to the Coordinator table, which may contain different transaction distributions over groups. Consequently, `com.scalar.db.transaction.consensuscommit.TwoPhaseConsensusCommit.commit()` must be called first by the coordinator service before being called by participant services.
```

Reply:
`ongoing transactions` might be better?

Reply:
`ongoing transactions` sounds good to me! Thanks!